### PR TITLE
Print PoS banner when terminal block is reached

### PR DESCRIPTION
**Pull request by asdacap**
Print PoS banner when terminal block is reached.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing

Synched kiln. Printed the following:
```
2022-06-13 18:54:42.6129|Beacon Headers from block 50001 to block 641075 | 434112 / 591075 | queue 38592 | current     0.00bps | total  5999.93bps 
2022-06-13 18:54:43.6132|Beacon Headers from block 50001 to block 641075 | 434112 / 591075 | queue 38592 | current     0.00bps | total  5918.07bps 
2022-06-13 18:54:43.6729|┌──────────────────────────────────────────────────────────────────────────────────────────┐ 
2022-06-13 18:54:43.6729|│ You have now reached...                                                                  │ 
2022-06-13 18:54:43.6729|│                   ,,,         ,,,                                                        │ 
2022-06-13 18:54:43.6729|│                 ;"   ^;     ;'   ",                                                      │ 
2022-06-13 18:54:43.6729|│                 ;    s$$$$$$$s     ;   ████████╗             ████████╗                   │ 
2022-06-13 18:54:43.6729|│                 ,  ss$$$$$$$$$$s  ,'   ██╔════██╗            ██╔═════╝                   │ 
2022-06-13 18:54:43.6729|│                 ;s$$$$$$$$$$$$$$$      ██║    ██║            ██║                         │ 
2022-06-13 18:54:43.6729|│                 $$$$$$$$$$$$$$$$$$     ██║    ██║  ███████╗  ██║                         │ 
2022-06-13 18:54:43.6729|│                $$$$P""Y$$$Y""W$$$$$    ████████╔╝ ██╔════██╗ ████████╗                   │ 
2022-06-13 18:54:43.6729|│                $$$$  p"$$$"q  $$$$$    ██╔═════╝  ██║    ██║ ╚═════██║                   │ 
2022-06-13 18:54:43.6729|│                $$$$  .$$$$$.  $$$$     ██║        ██║    ██║       ██║                   │ 
2022-06-13 18:54:43.6729|│                 $$DcaU$$$$$$$$$$       ██║        ██║    ██║       ██║                   │ 
2022-06-13 18:54:43.6729|│                   "Y$$$"*"$$$Y"        ██║        ╚███████╔╝ ████████║                   │ 
2022-06-13 18:54:43.6729|│                       "$b.$$"          ╚═╝         ╚══════╝  ╚═══════╝                   │ 
2022-06-13 18:54:43.6729|│                                                                                          │ 
2022-06-13 18:54:43.6729|│ Terminal block number: 55127                                                             │ 
2022-06-13 18:54:43.6729|│ Terminal block hash: 0x0d8429901f32846ee3a6b5765603e38cbcb8507c72a05aaaa681fad1876eba9e  │ 
2022-06-13 18:54:43.6729|└──────────────────────────────────────────────────────────────────────────────────────────┘ 
2022-06-13 18:54:44.6130|Beacon Headers from block 50001 to block 641075 | 434112 / 591075 | queue 38592 | current     0.00bps | total  5838.45bps 
```

## Further comments (optional)

Can't make nethermind logo fit well.

### CODE REVIEW

**Review by MarekM25 — CHANGES_REQUESTED** [review]
I think it is not a good place for this log. It is connected a bit with this PR https://github.com/NethermindEth/nethermind/pull/4145 and our bug on ropsten merge.
In geth, we have two logs:
Left PoW -> when we reached TTD
PoS activated -> when we received the first fcU (maybe on NewPayload too?)

Could you check https://eips.ethereum.org/EIPS/eip-3675 POS_FORKCHOICE_UPDATED in this EIP and consult with @marcindsobczak and @LukaszRozmej 

https://github.com/NethermindEth/nethermind/pull/4145 maybe a better solution than proposed in this PR will be having a flag (PoSActivated) on the first fcu and starting from this event not processing terminal blocks via gossip https://github.com/NethermindEth/nethermind/blob/merge/transition_fixes/src/Nethermind/Nethermind.Synchronization/SyncServer.cs#L257 

Watch out: I have an impression that the first fcU in hive tests is terminal block. but in real CL it is firstPoSBlock. It would be good to confirm with Mikhail what expected behavior is

File: src/Nethermind/Nethermind.Merge.Plugin/PoSSwitcher.cs [hunk 1]
```diff
@@ -155,13 +155,37 @@ public bool TryUpdateTerminalBlock(BlockHeader header)
             {
                 TerminalBlockReached?.Invoke(this, EventArgs.Empty);
                 _hasEverReachedTerminalDifficulty = true;
-                if (_logger.IsInfo) _logger.Info($"Reached terminal block {header}");
+                PrintPoSBanner(header);
```
Comment (marcindsobczak) [inline src/Nethermind/Nethermind.Merge.Plugin/PoSSwitcher.cs:158]: Printing banner in this place is wrong, maybe let's revert this change and only add the log to the codebase, without calling the method for now?

Comment (marcindsobczak) [reply src/Nethermind/Nethermind.Merge.Plugin/PoSSwitcher.cs:158]: Yes, here we just received terminal block for the first time (last PoW block), but there can be a few competing terminal blocks and PoS is starting when we receive first FCU (or NewPayload, not sure)

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Merge.Plugin/PoSSwitcher.cs:158]: Yeah terminal is POTENTIALLY last PoW block, but they can compete.

File: src/Nethermind/Nethermind.Merge.Plugin/PoSSwitcher.cs [hunk 2]
```diff
@@ -155,13 +155,37 @@ public bool TryUpdateTerminalBlock(BlockHeader header)
...
+            _logger.Info("│                $$$$P\"\"Y$$$Y\"\"W$$$$$    ████████╔╝ ██╔════██╗ ████████╗                   │");
+            _logger.Info("│                $$$$  p\"$$$\"q  $$$$$    ██╔═════╝  ██║    ██║ ╚═════██║                   │");
+            _logger.Info("│                $$$$  .$$$$$.  $$$$     ██║        ██║    ██║       ██║                   │");
+            _logger.Info("│                 $$DcaU$$$$$$$$$$       ██║        ██║    ██║       ██║                   │");
+            _logger.Info("│                   \"Y$$$\"*\"$$$Y\"        ██║        ╚███████╔╝ ████████║                   │");
+            _logger.Info("│                       \"$b.$$\"          ╚═╝         ╚══════╝  ╚═══════╝                   │");
+            _logger.Info("│                                                                                          │");
+            _logger.Info($"│ Terminal block number: {header.Number,-32}                                  │");
+            _logger.Info($"│ Terminal block hash: {header.Hash,-64}  │");
+            _logger.Info("└──────────────────────────────────────────────────────────────────────────────────────────┘");
```
[the remark this answers is not in the document]
Comment (asdacap) [reply src/Nethermind/Nethermind.Merge.Plugin/PoSSwitcher.cs:186]: I just like them indented at the same line.

**Review by LukaszRozmej — CHANGES_REQUESTED** [review]

**Review by MarekM25 — APPROVED** [review]

**Comment by MarekM25** [comment]
I think we didin't invoke this method @asdacap :)

**Review by MarekM25 — CHANGES_REQUESTED** [review]
Why not invoked? :) Could we invoke it as I said before?

**Comment by asdacap** [comment]
I still don't know where to invoke it. Last suggestion was during the first FCU, which did not get triggered.
